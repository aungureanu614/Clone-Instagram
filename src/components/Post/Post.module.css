.storyHeader {
  padding: 5px;
  display: flex;
}

.nickname {
  align-self: center;
  font-size: 14px;
  padding: 0 10px;
}

.caption {
  font-size: 14px;
}

.storyContainer {
  border: 1px solid lightgray;
  margin: 40px 0;
}