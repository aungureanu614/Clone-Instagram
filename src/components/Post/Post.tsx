import React from 'react'
import styles from './Post.module.css'
import { readdirSync } from 'fs'

interface Props {
  nickname: string
  avatar: React.ReactChild
  caption: string
  image: React.ReactChild
}

const Post: React.FunctionComponent<Props> = ({
  nickname,
  avatar,
  caption,
  image,
}) => (
  <div className={styles.storyContainer}>
    <div className={styles.storyHeader}>
      {avatar}
      <span className={styles.nickname}>
        <strong>{nickname}</strong>
      </span>
    </div>
    {image}
    <div className={styles.storyHeader}>
      <span className={styles.nickname}>
        <strong>{nickname}</strong>
      </span>
      <span className={styles.caption}>{caption}</span>
    </div>
  </div>
)

export default Post
