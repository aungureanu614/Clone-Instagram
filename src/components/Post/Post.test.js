import React from 'react'
import renderer from 'react-test-renderer'
import Post from './Post'

test('Post renders', () => {
  const component = renderer.create(
    <Post
      nickname="Post"
      avatar="http://placekitten.com/200/300"
      caption="this is a kitty"
      image="http://placekitten.com/200/300"
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
