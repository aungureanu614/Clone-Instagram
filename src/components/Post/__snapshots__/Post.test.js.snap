// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders 1`] = `
<div
  className="storyContainer"
>
  <div
    className="storyHeader"
  >
    http://placekitten.com/200/300
    <span
      className="nickname"
    >
      <strong>
        Post
      </strong>
    </span>
  </div>
  http://placekitten.com/200/300
  <div
    className="storyHeader"
  >
    <span
      className="nickname"
    >
      <strong>
        Post
      </strong>
    </span>
    <span
      className="caption"
    >
      this is a kitty
    </span>
  </div>
</div>
`;
